src/ui/components/JEXLDebuggerPage.tsx

```diff
@@ -10,17 +10,42 @@ import { Container, Row, Col, Form, Button } from "react-bootstrap";
 
 import { useToastsContext } from "../hooks/useToasts";
 import { evaluateJexl } from "../jexlParser";
+import ContextField from "./ContextField";
+
+type ContextValue = object | string | boolean | number | Date;
+
+function debounce<F extends (...args: any[]) => void>(
+  callback: F,
+  delay: number,
+): (...args: Parameters<F>) => void {
+  let timer: ReturnType<typeof setTimeout>;
+  return function (...args: Parameters<F>): void {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 const JEXLDebuggerPage: FC = () => {
-  const [clientContext, setClientContext] = useState({});
+  const [originalContext, setOriginalContext] = useState<
+    Record<string, ContextValue>
+  >({});
+  const [modifiedContext, setModifiedContext] = useState({});
+  const [formData, setFormData] = useState({});
   const [jexlExpression, setJexlExpression] = useState("");
   const [output, setOutput] = useState("");
   const { addToast } = useToastsContext();
 
   const fetchClientContext = useCallback(async () => {
+    setOriginalContext({});
+    setModifiedContext({});
+    setFormData({});
+
     try {
       const context = await browser.experiments.nimbus.getClientContext();
-      setClientContext(context);
+      setOriginalContext(context as Record<string, ContextValue>);
+      setModifiedContext(context);
```

Comment:
We don't need to do this. The modifiedContext only needs to contain overrides from the formData.

We should, however, serialize everything from context into formData at this point.

Reply:
Since we do 
```
const result = await evaluateJexl(jexlExpression, {
          ...originalContext,
          ...modifiedContext,
 });
```
for the JEXL Evaluation, I noticed that if we don't do `setModifiedContext(context);` after the refresh, then the evaluator outputs outdated information that doesn't match what the user sees in the context. Do you think I should still remove it and find a different workaround or did you have something else in mind?